### PR TITLE
PACA-1004: random claim generator improvements

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-1004](https://jira.cms.gov/browse/PACA-1004)

**User Story or Bug Summary:**

As a PACA engineer I want to have more control over random claim generation when testing the ETL pipeline.

---

### What Does This PR Do?

Adds new features to the random claim generators to facilitate their use in pipeline testing.  Each feature is configurable via command line arguments when running a server from the command line.

The new command line options are:

- `random.errorRate` sets the `RandomClaimGeneratorConfig.randomErrorRate`.  This is the number of claims per forced error.  I.e. 5 means approximately 20% of claims should have errors added to them.
- `random.max.mbi` sets the `RandomClaimGeneratorConfig.maxUniqueMbis`.  This is the maximum number of unique MBI's to be used for all claims.  Useful to force a smaller set of unique values when testing.
- `random.max.claimId` sets the `RandomClaimGeneratorConfig.maxUniqueClaimIds`.  This is the maximum number of unique claim id values to be used for all claims.  Smaller numbers generate more updates per claim which is useful in testing whether claim updates are processed correctly as well as limiting the size of the test database.
- `random.verbose` sets the `RandomClaimGeneratorConfig.optionalOverride`.  Setting this to true causes all optional fields have have a value in the generated claims.

Other changes include:

- Removes redundant uses of `/** {@link inheritDoc} */` when javadoc would insert the parent documentation for a method anyway.  `inheritDoc` should only be used when the comment also includes additional, specific documentation about the overridden version of the method.
- Fixes overly broad wildcard in root `pom.xml` that disabled generation of javadoc for nearly all classes in the RDA pipeline projects.
- Updates field lengths for claim id fields to use the full number of characters.  Previously they used shorter field lengths to reduce the number of unique ids and allow updates.  Now that can be done using the new configuration settings.
- Improves `MessageSource.skip` implementations when working with random claim generator.  Makes the skip instantaneous instead of using a loop and allows the forced error generation to be independent of sequence number.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java [hunk 1]
```diff
@@ -134,29 +133,22 @@ public class RandomFissClaimGenerator extends AbstractRandomClaimGenerator<FissC
...
-  /**
-   * Creates an instance with the specified seed.
-   *
-   * @param seed seed for the PRNG
-   */
-  public RandomFissClaimGenerator(long seed) {
-    super(seed, false, Clock.systemUTC());
-  }
+  /** Max length of a FISS claim id. */
+  private static final int FissClaimIdLength = 32;
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:137]: Is this a kotlin naming convention?

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java [hunk 2]
```diff
@@ -646,4 +641,27 @@ private void addRandomRevenueLines(FissClaim.Builder claim) {
...
+    }
+    if (shouldInsertErrorIntoCurrentClaim()) {
+      // update a random field with a string that is too long
+      oneOf(
+          "random-error",
+          () -> claim.setDcn(randomDigit(ForcedErrorFieldLength, ForcedErrorFieldLength)),
+          () ->
+              claim.setIntermediaryNb(randomDigit(ForcedErrorFieldLength, ForcedErrorFieldLength)));
+    }
+  }
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:666]: Another thought to add random errors could be to mess with the values at the level they are generated, for instance, `randomString(minLength, maxLength)` would return a string that is `maxLength + 1` based on the error chance, then the errors will scatter around different properties

**Review by KyRobbins — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java (hunk 2 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:666]: That interferes with the normal flow and spreads the logic around.  This way it's centralized.  All we need is an error to halt the pipeline.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java (hunk 1 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:137]: Nothing new there.  It's far more readable and keeps the names shorter so google format doesn't insert so many line breaks.

**Review by daryll — DISMISSED** [review]
lgtm !

**Review by cbrunefearless — APPROVED** [review]
lgtm

**Review by KyRobbins — APPROVED** [review]

**Review by cbrunefearless — APPROVED** [review]
lgtm

**Pull request merged**